src/box/relay.cc

```diff
@@ -1150,6 +1161,26 @@ relay_send_row(struct xstream *stream, struct xrow_header *packet)
 			say_warn("injected broken lsn: %lld",
 				 (long long) packet->lsn);
 		}
+		if (iproto_type_is_promote_request(packet->type)) {
+			struct synchro_request req;
+			xrow_decode_synchro(packet, &req);
+			/*
+			 * Promote must be sent only after corresponding RAFT
+			 * term was already sent. Because RAFT term is sent
+			 * from TX thread (not from wal), we process
+			 * tx_endpoint until corresponding RAFT term is sent.
```

Comment:
The problem with the comment is that it just narrates the code. It says what is done and I can see that in the code, which makes the comment not very useful. Usually a comment should answer the questions "why", not "what". What is it needed to send Raft term before PROMOTE/DEMOTE?